external/kafka/src/test/scala/org/apache/spark/streaming/kafka/KafkaStreamSuite.scala

```diff
@@ -65,12 +67,14 @@ class KafkaStreamSuite extends SparkFunSuite with Eventually with BeforeAndAfter
 
     val stream = KafkaUtils.createStream[String, String, StringDecoder, StringDecoder](
       ssc, kafkaParams, Map(topic -> 1), StorageLevel.MEMORY_ONLY)
-    val result = new mutable.HashMap[String, Long]() with mutable.SynchronizedMap[String, Long]
+    val result = new ConcurrentHashMap[String, Long].asScala
     stream.map(_._2).countByValue().foreachRDD { r =>
       val ret = r.collect()
       ret.toMap.foreach { kv =>
-        val count = result.getOrElseUpdate(kv._1, 0) + kv._2
-        result.put(kv._1, count)
+        result.synchronized {
+          val count = result.getOrElseUpdate(kv._1, 0) + kv._2
```

Comment:
@holdenk @zsxwing 
I tried  _val count = result.putIfAbsent(kv.1, 0) + kv._2,_ but the test failed for me.  So I will change to mutable.HashMap and put in synchronized block. 
Is it OK to use  mutable.HashMap and synchronized block in this file only, but use java.util.concurrent.ConcurrentHashMap in other files(StreamingListenerSuite, KinesisStreamTests and FileInputDStream)? Or is it better to  to use  mutable.HashMap and synchronized block for all the files that has SynchronizedMap?